### DISCUSSION TITLE
3D surface of ODF 2D plots

### DISCUSSION THREAD

**Discussion by 2dkcani** [mtex-toolbox/mtex#899] [Ask Anything]
Hi everybody,
I was wondering if there is a neat way of having 3D surfaces (random density in the z direction) out of the 2D sections of the ODFs in the Euler space?

BR,
Umberto.

**Comment by ralfHielscher** [comment]
Maybe you can start from here:
```matlab
% some example ODF
odf = SantaFe

% define a section
oS = phi2Sections(odf.CS,odf.SS,'phi2',30*degree);

% generate the grid
ori = oS.makeGrid;

% evaluate the odf
z = odf.eval(ori);

% plot it
surf(z)
```
Ralf.
